proxy/src/telemetry/metrics/histogram.rs

```diff
@@ -77,23 +80,46 @@ impl<V: Into<u64>> Histogram<V> {
             .expect("all values must fit into a bucket");
 
         self.buckets[idx].incr();
-        self.sum += Wrapping(value);
+        self.sum += value;
     }
+}
+
+impl<V: Into<u64>> FmtMetric for Histogram<V> {
+    fn fmt_metric<N: Display>(&self, f: &mut fmt::Formatter, name: N) -> fmt::Result {
+        let mut total = Counter::default();
+        for (le, count) in self.bounds.0.iter().zip(self.buckets.iter()) {
+            total += *count;
+            total.fmt_metric_labeled(f, Key(&name, "bucket"), format!("le=\"{}\"", le))?;
+        }
+        total.fmt_metric(f, Key(&name, "count"))?;
+        self.sum.fmt_metric(f, Key(&name, "sum"))?;
 
-    pub fn sum(&self) -> u64 {
-        self.sum.0
+        Ok(())
+    }
+
+    fn fmt_metric_labeled<N, L>(&self, f: &mut fmt::Formatter, name: N, labels: L) -> fmt::Result
+    where
+        N: Display,
+        L: Display,
+    {
+        let mut total = Counter::default();
+        for (le, count) in self.bounds.0.iter().zip(self.buckets.iter()) {
```

Comment:
TIOLI: you can dereference `count` in the `for` rather than in the body, like 
```rust
for (&le, count) in self {
    total += count;
    // ...
}
```
Also, as above, IIRC you can just use `for (le, count) in self` because `Histogram` has an `IntoIter` impl.